frmts/tiledb/tiledbdataset.cpp

```diff
@@ -646,6 +568,79 @@ TileDBDataset::~TileDBDataset()
     CSLDestroy(papszAttributes);
 }
 
+/************************************************************************/
+/*                           IRasterio()                                */
+/************************************************************************/
+
+CPLErr TileDBDataset::IRasterIO(GDALRWFlag eRWFlag, int nXOff, int nYOff,
+                                int nXSize, int nYSize, void *pData,
+                                int nBufXSize, int nBufYSize,
+                                GDALDataType eBufType, int nBandCount,
+                                int *panBandMap, GSpacing nPixelSpace,
+                                GSpacing nLineSpace, GSpacing nBandSpace,
+                                CPL_UNUSED GDALRasterIOExtraArg *psExtraArg)
+
+{
+    // support special case of writing attributes for bands, all attributes have to be set at once
+    const int nBufferDTSize(GDALGetDataTypeSizeBytes(eBufType));
+
+    if (eIndexMode == ATTRIBUTES && nBandCount == nBands &&
+        eBufType == eDataType && nXSize == nBufXSize && nYSize == nBufYSize &&
+        nBufferDTSize > 0 && (nPixelSpace % nBufferDTSize) == 0 &&
+        (nLineSpace % nBufferDTSize) == 0)
+    {
+        std::unique_ptr<tiledb::Query> poQuery;
+        std::vector<uint64_t> oaSubarray = {
+            (uint64_t)nYOff, (uint64_t)nYOff + nYSize - 1, (uint64_t)nXOff,
+            (uint64_t)nXOff + nXSize - 1};
+
+        if ((eRWFlag == GF_Read) && ((eAccess == GA_Update) && (m_roArray)))
```

Comment:
the (a) && ((b) && (c)) pattern looks suspicous to me. If that's intended the extra parenthesis around ((b) && (c)) should be removed. But I'm wondering if the first && shouldn't rather be a || ?

Reply:
I have removed the extra parenthesis. The logic is correct, the array can only be open in `read` or `write` mode, this is the case where we are doing an update and we read the data first and then apply the update.